transport/http/transport.go

```diff
@@ -97,19 +119,31 @@ func NewTransport(opts ...TransportOption) *Transport {
 }
 
 func buildClient(cfg *transportConfig) *http.Client {
-	return &http.Client{
-		Transport: &http.Transport{
-			// options lifted from https://golang.org/src/net/http/transport.go
-			Proxy: http.ProxyFromEnvironment,
-			Dial: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: cfg.keepAlive,
-			}).Dial,
-			TLSHandshakeTimeout:   10 * time.Second,
-			ExpectContinueTimeout: 1 * time.Second,
-			MaxIdleConnsPerHost:   cfg.maxIdleConnsPerHost,
-		},
+	transport := &http.Transport{
+		// options lifted from https://golang.org/src/net/http/transport.go
+		Proxy: http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: cfg.keepAlive,
+		}).Dial,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		MaxIdleConnsPerHost:   cfg.maxIdleConnsPerHost,
+	}
+
+	// Explicit passing this option in will override the ProxyFromEnvironment
+	if cfg.socks5Proxy != "" {
+		dialer, err := proxy.SOCKS5("tcp", cfg.socks5Proxy, nil, proxy.Direct)
+		if err != nil {
+			log.Fatalf("Error creating SOCKS proxy: %v", err)
```

Comment:
As we discussed, I wound up doing the proxy connection in the Start() method because it is the first place errors are returned. Updated in next patch.